yarn-project/end-to-end/src/spartan/add_additional_validators.test.ts

```diff
@@ -0,0 +1,311 @@
+        resource: `svc/${config.INSTANCE_NAME}-aztec-network-eth-execution`,
+        namespace: config.NAMESPACE,
+        containerPort: config.CONTAINER_ETHEREUM_PORT,
+      });
+      forwardProcesses.push(process);
+      ETHEREUM_HOSTS = `http://127.0.0.1:${port}`;
+    }
+    pxe = await createCompatibleClient(PXE_URL, debugLogger);
+  });
+
+  afterAll(() => {
+    forwardProcesses.forEach(p => p.kill());
+  });
+
+  const generateValidatorAddresses = (numberOfValidators: number) => {
+    const keys: string[] = [];
+    const accounts: Account[] = [];
+    for (let i = 0; i < numberOfValidators; i++) {
+      const key = generatePrivateKey();
+      const account = privateKeyToAccount(key);
+      keys.push(key);
+      accounts.push(account);
+    }
+    return { keys, accounts };
+  };
+
+  // We assume that the base account is already funded
+  const fundAccount = async (walletClient: ViemWalletClient, account: Account) => {
+    await walletClient.sendTransaction({
+      to: account.address,
+      value: parseEther('100'),
+    });
+  };
+
+  const addValidatorViaCli = async (rollupAddress: string, account: Account) => {
+    const validatorAddress = account.address;
+
+    const exitCode = await runAztecBin(
+      [
+        'add-l1-validator',
+        '--validator',
+        validatorAddress,
+        '--mnemonic',
+        config.L1_ACCOUNT_MNEMONIC,
+        '--rollup',
+        rollupAddress,
+        '--l1-chain-id',
+        '1337',
+        '--l1-rpc-urls',
+        ETHEREUM_HOSTS,
+      ],
+      debugLogger,
+    );
+    expect(exitCode).toBe(0);
+  };
+
+  /**
+   * Deploy the additional validators using the helm chart
+   * @param keys - The private keys of the validators to deploy
+   */
+  const deployAdditionalValidators = async (
+    validatorAddresses: string[],
+    keys: string[],
+    enr: string,
+    registryAddress: string,
+    extraEnv: Partial<Record<EnvVar, string>>,
+  ) => {
+    const validatorContainers = [];
+    for (let i = 0; i < validatorAddresses.length; i++) {
+      const validator = await deployAdditionalValidator(
+        i,
+        validatorAddresses[i],
+        keys[i],
+        enr,
+        registryAddress,
+        extraEnv,
+      );
+      validatorContainers.push(validator);
+    }
+    return validatorContainers;
+  };
+
+  const deployAdditionalValidator = async (
+    index: number,
+    validatorAddress: string,
+    key: string,
+    enr: string,
+    registryAddress: string,
+    extraEnv: Partial<Record<EnvVar, string>>,
+  ) => {
+    return await kubectlRun(
+      // Container name
+      `additional-validator-${index}`,
+      // Namespace
+      config.NAMESPACE,
+      // Image tag
+      `aztecprotocol/aztec:${config.AZTEC_DOCKER_TAG}`,
+      // Environment variables
+      {
+        // TODO: include in config
+        ...extraEnv,
+        LOG_LEVEL: 'debug',
+        ETHEREUM_SLOT_DURATION: config.ETHEREUM_SLOT_DURATION.toString(),
```

Comment:
Would be interested to see if these `ETHEREUM_` and `AZTEC_` vars can be removed as they should all be read from L1.

Reply:
Ah yessir that is true